Source/WebCore/rendering/BaselineAlignment.cpp

```diff
@@ -88,6 +88,11 @@ const BaselineGroup& BaselineContext::sharedGroup(const RenderBox& child, ItemPo
     return const_cast<BaselineContext*>(this)->findCompatibleSharedGroup(child, preference);
 }
 
+Vector<BaselineGroup> BaselineContext::sharedGroups() const
```

Comment:
this seems to return a _copy_ of the vector. is that what callers expect/need here?

Reply:
Nope this should return a reference. Also had to make `baselineContext` in `LineContext` mutable since the alignment code uses `const LineContext`. I think this is better than removing the `const` from the function argument, but let me know if you think we should do that instead or take another approach for that

Reply:
"context" to me reads like something you don't mutate but instead you change the associated "state". Maybe it's just incorrectly named structs/classes, not sure, but I think even in the commit message you referred to some of these things as "state". 
Seeing a single member of a "context" class mutable (e.g. while debugging some issues) would certainly make me scratch my head. :)

Reply:
Yeah so I removed the mutable from the now `baselineState` and now whenever we need to modify it I also removed the const from the `LineState`